### PR TITLE
feat: add Cloud Code context to `ParseObject.fetch`

### PR DESCRIPTION
**Pull request by yog27ray**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

This extension the of the https://github.com/parse-community/parse-server/pull/6626. To pass context while making fetch call.

### TODOs before merging

- [x] Add tests

### CODE REVIEW

**Review by mtrezza — COMMENTED** [review]
It seems that this adds the context option to the `ParseObject.fetch` method in general, so it's not just related to `afterFind`, but it also works for `beforeFind`, correct?

**Comment by yog27ray** [comment]
@mtrezza yes, the same context will be passed to beforeFind and then to afterFind also.

**Comment by mtrezza** [comment]
Great, just wanted to make sure to name the PR title correctly for the changelog entry. Could you add `beforeFind` to the existing test?

File: spec/CloudCode.spec.js [hunk 1]
```diff
@@ -3224,6 +3224,21 @@ describe('afterLogin hook', () => {
     const query = new Parse.Query(TestObject);
     await query.find({ context: { a: 'a' } });
   });
+
+  it('beforeFine and afterFind should have access to context while making fetch call', async () => {
```
Comment (mtrezza) [inline spec/CloudCode.spec.js]: typo `beforeFind`

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-alpha.21](https://github.com/parse-community/parse-server/releases/tag/5.0.0-alpha.21)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.10](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.10)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0](https://github.com/parse-community/parse-server/releases/tag/5.1.0)

**Pull request merged**